wire-ios-sync-engine/Source/SessionManager/SessionManager.swift

```diff
@@ -641,23 +641,28 @@ public final class SessionManager: NSObject, SessionManagerType {
         return account
     }
 
-    private func selectInitialAccount(_ account: Account, launchOptions: LaunchOptions) {
+    private func selectInitialAccount(
+        _ account: Account,
+        launchOptions: LaunchOptions,
+        completion: @escaping () -> Void
+    ) {
         if let url = launchOptions[UIApplication.LaunchOptionsKey.url] as? URL {
             if (try? URLAction(url: url))?.causesLogout == true {
                 // Do not log in if the launch URL action causes a logout
-                return
+                return completion()
```

Comment:
do we need to enable tracking yet if the user logged out? Is it useful for the caller to know if it was successful or not. In that case we could pass a bool in the completion

Reply:
> do we need to enable tracking yet if the user logged out?

@netbe No, which brings the question do we need the completion there after all. cc: @caldrian

Reply:
@agisilaos well I guess you wanted to execute the tracking after the session is loaded

Reply:
That was the intention yes, because fore that it wouldn't be possible.

Reply:
I'm not sure we handle login/logout/multiple accounts gracefully.

Reply:
I already done that on a previous ticket if I remember correctly. Making sure the analytics user profile is updated and we start a new session if there's a different user e.g has multiple accounts.